src/winetricks

```diff
@@ -1088,45 +1119,53 @@ w_download_to()
                 case "$_W_shatype" in
                     none)
                         w_warn "No checksum provided, not verifying"
-                        ;;
+                    ;;
                     sha256)
                         w_get_sha256sum "$_W_cache/$_W_file"
-                        if [ "$_W_gotsha256sum"x = "$_W_sum"x ] ; then
+                        if [ "$_W_gotsha256sum" = "$_W_sum" ]; then
                             checksum_ok=1
                             break
                         fi
-                        ;;
+										;;
+										sha512)
+												w_get_sha512sum "$_W_cache/$_W_file"
+                        if [ "$_W_gotsha256sum" = "$_W_sum" ]; then
+                            checksum_ok=1
+                            break
+                        fi
+                    ;;
                 esac
 
-                if test ! "$WINETRICKS_CONTINUE_DOWNLOAD" ; then
+                if [ -n "$WINETRICKS_CONTINUE_DOWNLOAD" ]; then
                     case $LANG in
                         pl*) w_warn "Niezgodność sum kontrolnych dla $_W_cache/$_W_file, pobieram ponownie" ;;
                         ru*) w_warn "Контрольная сумма файла $_W_cache/$_W_file не совпадает, попытка повторной загрузки" ;;
                         *) w_warn "Checksum for $_W_cache/$_W_file did not match, retrying download" ;;
                     esac
                     mv -f "$_W_cache/$_W_file" "$_W_cache/$_W_file".bak
                 fi
-            else
+            elif [ -z "$_W_sum" ]; then
                 # file exists, no checksum known, declare success and exit loop
                 break
             fi
-        elif test -f "$_W_cache/$_W_file" ; then
+        elif [ -f "$_W_cache/$_W_file" ]; then
             # zero-length file, just delete before retrying
             rm "$_W_cache/$_W_file"
         fi
 
         w_try_cd "$_W_cache"
         # Mac folks tend to have curl rather than wget
         # On Mac, 'which' doesn't return good exit status
-        echo "Downloading $_W_url to $_W_cache"
+        printf '%s\n' "Downloading $_W_url to $_W_cache"
 
         # For sites that prefer Mozilla in the user-agent header, set W_BROWSERAGENT=1
         case "$W_BROWSERAGENT" in
             1) _W_agent="Mozilla/5.0 (compatible; Konqueror/2.1.1; X11)" ;;
             *) _W_agent="" ;;
         esac
 
-        if [ "${WINETRICKS_DOWNLOADER}" = "aria2c" ] ; then
+				# Sanitized to allow mirrors
+        if [ "$WINETRICKS_DOWNLOADER" = aria2c ] ; then
```

Comment:
Why?

Reply:
I believe that it's useless use of curly brackets